src/telcameras_v3/ingress_parser.py

```diff
@@ -28,17 +26,50 @@ def get_default_batch_size(self):
         return 10000
 
     def consume_raw_data(self, raw_data):
-        observation = json.loads(raw_data)
-        observation['message_id'] = observation.pop('id')
-        observation['sensor_state'] = observation.pop('status')
-        observation['groupaggregates'] = observation.pop('direction')
-        for groupaggregate in observation['groupaggregates']:
-            groupaggregate['observation_timestamp'] = observation['timestamp']
-            groupaggregate['persons'] = groupaggregate.pop('signals', [])
-            for person in groupaggregate['persons']:
-                person['person_observation_timestamp'] = person.pop('observation_timestamp')
-                person['observation_timestamp'] = observation['timestamp']
-
+        observation = self.data_to_observation(raw_data)
         observation_serializer = ObservationSerializer(data=observation)
         observation_serializer.is_valid(raise_exception=True)
         return observation_serializer.save()
+
+    def data_to_observation(self, raw_data: dict):
+        raw_json = json.loads(raw_data)
+        sensor_name = raw_json["sensor"]
+        observation = dict(
+            message_id=raw_json["id"],
+            timestamp=raw_json["timestamp"],
+            sensor_name=sensor_name,
+            sensor_type=raw_json["sensor_type"],
+            sensor_state=raw_json["status"],
+            # sensor=sensor_name,
+            area=sensor_name,
+            latitude=raw_json["latitude"],
+            longitude=raw_json["longitude"],
+            interval=raw_json["interval"],
+            density=raw_json["density"],
+            groupaggregates=[
+                dict(
+                    line=f"{sensor_name}-{aggregate['azimuth']}",
+                    observation_timestamp=raw_json["timestamp"],
+                    azimuth=aggregate["azimuth"],
+                    count=aggregate["count"],
+                    cumulative_distance=aggregate["cumulative_distance"],
+                    cumulative_time=aggregate["cumulative_time"],
+                    median_speed=aggregate["median_speed"],
+                    count_scrambled=None,  # This field is filled by the serializer
+                    persons=[
+                        dict(
+                            observation_timestamp=raw_json["timestamp"],
+                            person_observation_timestamp=person["observation_timestamp"],
+                            record=person["record"],
+                            distance=person["distance"],
+                            time=person["time"],
+                            speed=person["speed"],
+                            type=person["type"],
+                        )
+                        for person in aggregate.get("signals", [])
+                    ],
+                )
+                for aggregate in raw_json["direction"]
+            ],
```

Comment:
Vanwaar deze meer langere versie hiervan? Wat zijn de voordelen ervan? Ik zie namelijk ook nadelen, zoals dat wanneer er een model change is die ook hier expliciet veranderd moet worden.

Reply:
In de oude vorm worden er vanuit de aangeleverde json velden aan de serializer doorgegeven zonder dat deze velden waar dan ook in de code genoemd worden. Dit maakt de code totaal onleesbaar voor iemand die nieuw binnenkomt op dit project. In deze 'langere' vorm wordt alles expliciet doorgegeven aan de serializer waardoor ik kan zien wat er in de code gebeurd. Mijn idee was dat onleesbare code leidt tot meer fouten. 

Als je een model change kunt doen zonder de code in de parser te hoeven updaten is dat in mijn ogen niet iets positiefs...